proto/babylonlabs/babylon/v1beta1/tx.proto

```diff
@@ -13,11 +13,61 @@ option (gogoproto.goproto_getters_all) = false;
 service Msg {
   option (cosmos.msg.v1.service) = true;
 
+  // InstantiateBabylonContracts defines an operation for instantiating the
+  // Babylon contracts.
+  rpc InstantiateBabylonContracts(MsgInstantiateBabylonContracts)
+      returns (MsgInstantiateBabylonContractsResponse);
+
   // UpdateParams defines a (governance) operation for updating the x/auth
   // module parameters. The authority defaults to the x/gov module account.
   rpc UpdateParams(MsgUpdateParams) returns (MsgUpdateParamsResponse);
 }
 
+// MsgInstantiateBabylonContracts is the Msg/InstantiateBabylonContracts request
+// type.
+message MsgInstantiateBabylonContracts {
+  option (cosmos.msg.v1.signer) = "signer";
+
+  // signer is the address who submits the message.
+  string signer = 1 [ (cosmos_proto.scalar) = "cosmos.AddressString" ];
+
+  // babylon_contract_code_id is the code ID for the Babylon contract.
+  uint64 babylon_contract_code_id = 2;
+  // btc_staking_contract_code_id is the code ID for the BTC staking contract.
+  uint64 btc_staking_contract_code_id = 3;
+  // btc_finality_contract_code_id is the code ID for the BTC finality contract.
+  uint64 btc_finality_contract_code_id = 4;
+
+  // network is the Bitcoin network to connect to (e.g. "regtest", "testnet",
+  // "mainnet")
+  string network = 5;
+  // babylon_tag is a unique identifier for this Babylon instance
+  string babylon_tag = 6;
+  // btc_confirmation_depth is the number of confirmations required for Bitcoin
+  // transactions
+  uint32 btc_confirmation_depth = 7;
+  // checkpoint_finalization_timeout is the timeout in blocks for checkpoint
+  // finalization
+  uint32 checkpoint_finalization_timeout = 8;
+  // notify_cosmos_zone indicates whether to notify the Cosmos zone of events
+  bool notify_cosmos_zone = 9;
+  // btc_staking_msg is the initialization message for the BTC staking contract
+  bytes btc_staking_msg = 10;
+  // btc_finality_msg is the initialization message for the BTC finality
+  // contract
+  bytes btc_finality_msg = 11;
+  // consumer_name is the name of this consumer chain
```

Comment:
The params here can be just hard-coded in the SDK IMO.

Reply:
Good point, this is something I'm not entirely sure. One thing that I don't know whether we should hard-code is the contract admin. Probably it's always be gov module account, or some security council accounts.

Will add a TODO for now

Reply:
The SDK, and in particular the demo / test apps, are just a way to organise and bootstrap things. There can be many others.